pkg/operator/sync_openshiftcontrollermanager_v311_00_test.go

```diff
@@ -25,6 +32,155 @@ import (
 	operatorv1helpers "github.com/openshift/library-go/pkg/operator/v1helpers"
 )
 
+type clusterVersionLister struct {
+}
+
+// List lists all ClusterVersions in the indexer.
+func (s *clusterVersionLister) List(selector labels.Selector) (ret []*v1.ClusterVersion, err error) {
+	return ret, err
+}
+
+// Get retrieves the ClusterVersion from the index for a given name.
+func (s *clusterVersionLister) Get(name string) (*v1.ClusterVersion, error) {
+	// should disable controllers that are known but not enabled
+	if name == "KnownButdisabled" {
+		return &v1.ClusterVersion{
+			Status: v1.ClusterVersionStatus{
+				Capabilities: v1.ClusterVersionCapabilitiesStatus{
+					KnownCapabilities: []v1.ClusterVersionCapability{
+						configv1.ClusterVersionCapabilityBuild,
+						configv1.ClusterVersionCapabilityDeploymentConfig,
+					},
+					EnabledCapabilities: []v1.ClusterVersionCapability{
+						// empty caps
+					},
+				},
+			},
+		}, nil
+	}
+	// should leave all controllers enabled if they are not known
+	if name == "Unknown" {
+		return &v1.ClusterVersion{
+			Status: v1.ClusterVersionStatus{
+				Capabilities: v1.ClusterVersionCapabilitiesStatus{
+					KnownCapabilities: []v1.ClusterVersionCapability{
+						// empty caps
+					},
+					EnabledCapabilities: []v1.ClusterVersionCapability{
+						// empty caps
+					},
+				},
+			},
+		}, nil
+	}
+	// should leave all controllers enabled if they are known and enabled
+	return &v1.ClusterVersion{
+		Status: v1.ClusterVersionStatus{
+			Capabilities: v1.ClusterVersionCapabilitiesStatus{
+				KnownCapabilities: []v1.ClusterVersionCapability{
+					configv1.ClusterVersionCapabilityBuild,
+					configv1.ClusterVersionCapabilityDeploymentConfig,
+				},
+				EnabledCapabilities: []v1.ClusterVersionCapability{
+					configv1.ClusterVersionCapabilityBuild,
+					configv1.ClusterVersionCapabilityDeploymentConfig,
+				},
+			},
+		},
+	}, nil
+}
+
+func TestConfigMapControllerDisabling(t *testing.T) {
+
+	testCases := []struct {
+		name          string
+		versionLister configlisterv1.ClusterVersionLister
```

Comment:
The current local implementation of `clusterVersionLister` makes is harder to read which capabilities are actually known/enabled. Let's define them directly for each test through
```
knownCapabilities       []configv1.ClusterVersionCapability
enabledCapabilities     []configv1.ClusterVersionCapability
```
as in https://github.com/openshift/cluster-openshift-apiserver-operator/pull/532/files#diff-25e7c9a2521d15c93e122ab635c1ec19358116650aa275abca5e014b514c69c2R239-R240.

Reply:
done